### ISSUE TITLE
Black Artefacts (Holes) in translated images

### ISSUE THREAD

**Issue by paviddavid** [junyanz/pytorch-CycleGAN-and-pix2pix#725]
Hi all, 

I working with this repo and i totally love it. However, I am concerned about the produced/translated results. Many test images look like the attached image and contain some black holes which should not be there compared to the real image. 

Does anybody has an idea what could be the reason for this strange behavior?

![0a0a0b1a-7c39d841_fake_B](https://user-images.githubusercontent.com/45428272/62560210-d9bcdf00-b87c-11e9-911f-a09582e6f1df.png)

Thanks in advance

**Comment by junyanz** [comment]
It sometimes happens for some applications. It's kind of mode collapse in the patch level and hard to fix directly. If you are using paired datasets, you can try more recent models such as [pix2pixHD](https://github.com/NVIDIA/pix2pixHD) and [SPADE](https://github.com/NVlabs/SPADE). If you are using unpaired datasets, you can try to increase identity loss and/or cycle consistency loss.

**Comment by paviddavid** [comment]
@junyanz  Thanks for your qick response! 

It is very interesting because I translated images from Cityscapes to Berkeley and vice versa. With the default learning rate (0.0002) everything looks fine but with the increased learning rate of 0.002 those strange artefacts show up. 

I am using unpaired data. How can I increase the loss? I am unsure what this means..

**Comment by vis-opt** [comment]
I think they mean try boosting the `lambda_A, lambda_B` arguments in `cycle_gan_model.py`. Default is 10, you can try 25 or 50 for example.

**Comment by junyanz** [comment]
A smaller learning rate will also make training more stable.

**Comment by paviddavid** [comment]
Thanks a lot for your fast responses. 
@vis-opt Maybe I am blind or whatever, but I do not find the `lambda_A ` and `lambda_B ` values in the option files. How is it possible to change the values? 

@junyanz Good point, I increased the learning rate and by `lr=0.0002` everything was fine, `lr=0.002` lead to these strange behaviours. So, I should test `lr=0.00002` to compare them. 

Do you have an idea if these black holes could make the png file "corrupt"? Further evaluations I wanted to run were not possible because of some files which are obviously corrupt. Do you think these black holes could be the reason for that?


Thanks a lot in advance. I really appreciate your

**Comment by junyanz** [comment]
`--lambda_A` and `--lambda_B` control the magnitude of the cycle-consistency loss. See this [line](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/models/cycle_gan_model.py#L41) for more details.

Not sure it black holes will make the file corrupt. As the generator's output is bounded by a tanh. It might happen if you have an inf or nan during training.
